docs/uml/sequenceDiagrams/miscellaneous/obtainAndParseUserInput.puml

```diff
@@ -1,11 +1,46 @@
 @startuml
 !include ../../Style.puml
 
+title Sequence Diagram Showing Receiving and Parsing User Input
+hide footbox
+
+autonumber
+
 participant "werkIt:WerkIt" AS WerkIt
 participant "ui:UI" AS UI
+participant "parser:Parser" AS Parser
+participant "newCommand\n:Command" AS newCommand
 
+activate WerkIt
 WerkIt -> UI : getUserInput()
 activate UI
 return userInputTrimmed:String
 
+WerkIt -> Parser : parseUserInput(userInputTrimmed)
+activate Parser
+ref over Parser
+Check for illegal characters
+end
+
+alt Command starts with 'workout'
+    Parser -> Parser : createWorkoutCommand(userInput)
+else Command starts with 'exit'
+    Parser -> Parser : createExitCommand(userInput)
+else Command starts with 'help'
+    Parser -> Parser : createHelpCommand(userInput)
+else Command starts with 'exercise'
+    Parser -> Parser : createExerciseCommand(userInput)
+else Command starts with 'search'
+    Parser -> Parser : createSearchCommand(userInput)
+else Command starts with 'plan'
+    Parser -> Parser : createPlanCommand(userInput)
+else Command starts with 'schedule'
+    Parser -> Parser : createScheduleCommand(userInput)
+end
+
+return newCommand:Command
+
+WerkIt -> newCommand : execute()
+activate newCommand
```

Comment:
Not sure if we should include the `execute()` method here as the other diagrams include this method too. It would look like it was called two times in a SD diagram.

Reply:
i agree with musfirah, perhaps the execute() could be excluded from the diagram

Reply:
Thanks for your feedback @Musfirahe0556596 and @emilysim00, I have removed the `execute()` method in the SD in Commit c3c0380. However, please advise if I should remove the statement on this in my write up in the developer guide too.

Reply:
Discussed on Telegram, we've decided to keep the write up in the DG to provide a link to the other sections.